File: Golang/LongestCommonPrefix/LongestCommonPrefix_test.go
package LongestCommonPrefix

import "testing"

func TestLongestCommonPrefix(t *testing.T) {

	if longestCommonPrefix([]string{}) == "" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if longestCommonPrefix([]string{"abc"}) == "abc" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if longestCommonPrefix([]string{"abcf", "abcd", "abcdefg"}) == "abc" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if longestCommonPrefix([]string{"aa", "a"}) == "a" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/LongestCommonPrefixTest.py
from unittest import TestCase

from LongestCommonPrefix import LongestCommonPrefix


class TestLongestCommonPrefix(TestCase):
    def test_longestCommonPrefix(self):
        lcp = LongestCommonPrefix()

        self.assertTrue(len(lcp.longestCommonPrefix([""])) == 0)

        self.assertEqual(lcp.longestCommonPrefix(["abc"]), "abc")

        self.assertEqual(lcp.longestCommonPrefix(["abcf", "abcd", "abcdefg"]), "abc")

        self.assertEqual(lcp.longestCommonPrefix(["aa", "a"]), "a")
